webapp/graphite/tags/utils.py

```diff
@@ -0,0 +1,94 @@
+"""Utility functions for tag databases."""
+import abc
+
+try:
+    from importlib import import_module
+except ImportError:  # python < 2.7 compatibility
+    from django.utils.importlib import import_module
+
+
+class TaggedSeries(object):
+  @classmethod
+  def parse(cls, path):
+    segments = path.split(';')
+
+    metric = segments[0]
+    if not metric:
+      raise Exception('Cannot parse path %s, no metric found' % path)
+
+    tags = {}
+
+    for segment in segments[1:]:
+      tag = segment.split('=', 1)
+      if len(tag) != 2 or not tag[0]:
```

Comment:
Our tag selector logic now matches the prometheus spec, and an empty value is considered valid.